src/lib/utils/inventoryUtils.ts

```diff
@@ -1,6 +1,29 @@
-import { InventoryRecord } from "../airtable/interface";
+import moment, { Moment } from 'moment';
+import { InventoryRecord, InventoryUpdateRecord, PurchaseRequestRecord } from '../airtable/interface';
+import {
+  selectInventoryUpdatesArrayByInventoryId,
+  selectPurchaseRequestsArrayByInventoryId
+} from '../redux/inventoryData';
+import { store } from '../redux/store';
 
 // Calculate when an inventory record was last updated
-export const getInventoryLastUpdated = (inventory: InventoryRecord) => {
-    return "00/00/00 00:00pm"; // TODO @wangannie: replace with real calculation
-}
\ No newline at end of file
+export const getInventoryLastUpdated = (inventory: InventoryRecord): string => {
+  const purchaseRequests: PurchaseRequestRecord[] = selectPurchaseRequestsArrayByInventoryId(
+    store.getState(),
+    inventory.id,
+  );
+  const inventoryUpdates: InventoryUpdateRecord[] = selectInventoryUpdatesArrayByInventoryId(
+    store.getState(),
+    inventory.id,
+  );
+
+  const updateTimes: Moment[] = [];
+  purchaseRequests.map((pr) => updateTimes.push(moment(pr.createdAt)));
+  inventoryUpdates.map((update) => updateTimes.push(moment(update.createdAt)));
+
+  // Check because moment will incorrectly return the current time if updateTimes is empty
+  if (updateTimes.length > 0) {
+    return moment.max(updateTimes).toLocaleString();
+  }
+  return 'Unknown'; // TODO @wangannie: address design edge case
```

Comment:
Hmm yeah I think it's just because by default moment will incorrectly return the current time if given an empty array, which would be very misleading if that were to ever happen for any reason (but I agree, no specific cases come to mind besides the airtable cases you mentioned). I guess this is just to make sure we don't ever let it just return the current time by default.